lucene/grouping/src/java/org/apache/lucene/search/grouping/FirstPassGroupingCollector.java

```diff
@@ -139,10 +139,18 @@ public ScoreMode scoreMode() {
       // System.out.println("  group=" + (group.groupValue == null ? "null" : group.groupValue.toString()));
       SearchGroup<T> searchGroup = new SearchGroup<>();
       searchGroup.groupValue = group.groupValue;
+      // We pass this around so that we can get the corresponding solr id when serializing the search group to send to the federator
+      searchGroup.topDocLuceneId = group.topDoc;
       searchGroup.sortValues = new Object[sortFieldCount];
       for(int sortFieldIDX=0;sortFieldIDX<sortFieldCount;sortFieldIDX++) {
         searchGroup.sortValues[sortFieldIDX] = comparators[sortFieldIDX].value(group.comparatorSlot);
       }
+      // TODO: It should be possible to extend this to handle more than one FieldComparator and other types
```

Comment:
TODO item here w.r.t. sort field count being one. If it's not too difficult extending to multiples might be nice to do, or it could be deferred until later (and `GroupingSpecification.validate` would for now throw if there's more than one). Either way let's add test coverage and documentation.

Semi-related, the https://lucene.apache.org/solr/guide/8_1/common-query-parameters.html#sort-parameter documentation reminded me that sorting by functions is possible. If (as the code snippet here seems to suggest) `group.skip.second.step=true` currently assumes that the sort element is a field (and not a function) then let's include that in the documentation, validation code and test coverage. Support for the not-yet-supported things could of course be added subsequently in future. Does that kind of make sense?

Reply:
I fixed the TODO - `topDocScore` seems to be used only to add the `maxScore` in the response - and I added tests that check that in distribute mode the values are the same with or without the skipping the second step.  

sort by function and document about that will be tacked in a separate comment.

Reply:
a5c5e616d3 3e4c9d0 forbid `group.func` (not supported by distribute search anyway) and `group.query` (prefer to do this in a separate moment because I think it will require some time to make sure that it works and test it). I also cover that into the documentation.